### DISCUSSION TITLE
Would adding a changeset tag for edits that were made far away be worthwhile?

### DISCUSSION THREAD

**Discussion by ygra** [streetcomplete/StreetComplete#4017] [Ideas]
I recently came across a number of garages (or something similar, perhaps even some sort of shed for garbage bins) in one area that have been tagged as `building=apartments` with StreetComplete and were wondering a bit how that came to be. My assumption would be that someone standing right in front of it wouldn't make that mistake, especially not multiple times. However, those buildings (if you disregard shadow lengths) look a lot like the apartment buildings next to it on aerial imagery and I guess someone may have just tagged from aerial images instead of on the ground.
Perhaps such suspicions could be validated easier (and perhaps more changes disregarded upon review) if the changeset includes information like »This changeset includes changes made from a considerable distance«.
I still believe the vast majority of StreetComplete changes to be valid and good, though. It's just one data point here where I noticed something amiss.

**Comment by mnalis** [comment]
Or remove `source=survey` changeset tag (as suggested in related https://github.com/streetcomplete/StreetComplete/pull/2794#issuecomment-894874562)

**Comment by matkoniecz** [comment]
A significant benefit would be making clear that it is mappers fault and misuse of StreetComplete, not a StreetComplete fault.

**Reply by mnalis** [reply]
see also: https://github.com/streetcomplete/StreetComplete/issues/4081

**Comment by matkoniecz** [comment]
I think that I would modify https://github.com/streetcomplete/StreetComplete/blob/18189f123416d9a0b332e50afab58b857452b416/app/src/main/java/de/westnordost/streetcomplete/screens/main/QuestSourceIsSurveyChecker.kt#L24 so that it would return 


`YES_ACCORDING_TO_GPS`, `YES_ACCORDING_TO_USER` and `NO`

In case of `YES_ACCORDING_TO_USER` I would set additional note to changeset, something like `user_not_directly_on_site_according_to_gps=yes`

**Reply by mnalis** [reply]
To note: it would probably be good idea to flush & close existing changeset on such state change, so the quests with different gps tags do no mix. (e.g. so `lit` quests which were with `YES_ACCORDING_TO_GPS` would end in one changeset with one tag, while `lit` quests which were `YES_ACCORDING_TO_USER` would end in another changeset with another tag).

If that is too complicated, then (as less useful alternative) at least "worst" of all statuses should be remembered (e.g. if I solved one quest which is `YES_ACCORDING_TO_GPS`, then `x` quests which are `YES_ACCORDING_TO_USER`, and then again one quest which is `YES_ACCORDING_TO_GPS`, whole changeset should always be marked as `YES_ACCORDING_TO_USER`, no matter if `x` was small or big)

There also might be potential issues to be aware  if SC does not manually close changesets but instead lets them expire/autoclose, and user is using auto-upload (e.g. changeset might have initially been opened with some values, and we want to change them if the situation changes).

P.S. what would be the difference  between `YES_ACCORDING_TO_USER` and `NO` ?

**Reply by matkoniecz** [reply]
> P.S. what would be the difference between YES_ACCORDING_TO_USER and NO ?

`NO` refuses to make edit, `YES_ACCORDING_TO_USER` allows to make edit but marks it as user claiming to be on location without GPS confirmation

> To note: it would probably be good idea to flush & close existing changeset on such state change, so the quests with different gps tags do no mix.

Why flushing and closing is needed?

>  at least "worst" of all statuses should be remembered

It is impossible to add tag on changeset: if not open already you need to open a new one. So "lets fit change to changeset with correct tags" is needed anyway
